lib/releases.js

```diff
@@ -237,17 +237,33 @@ const generateChangeLog = (mergedPullRequests, config) => {
     changeLog.push(pullRequestToString(uncategorizedPullRequests), '\n\n')
   }
 
-  categorizedPullRequests.map((category, index) => {
-    if (category.pullRequests.length > 0) {
+  for (const [index, category] of categorizedPullRequests.entries()) {
+    if (category.pullRequests.length === 0) {
+      continue
+    }
+    const categoryTitle = template(config['category-template'], {
+      $TITLE: category.title,
+    })
+    const pullRequestString = pullRequestToString(category.pullRequests)
+
+    if (category.pullRequests.length <= category['collapse-after']) {
+      changeLog.push(categoryTitle, '\n\n', pullRequestString)
+    } else {
       changeLog.push(
-        template(config['category-template'], { $TITLE: category.title }),
+        categoryTitle,
+        '\n\n',
+        '<details>',
+        '\n',
+        `<summary>${category.pullRequests.length} changes</summary>`,
```

Comment:
I think it would be best to swap the if and else statement around. According to my suggestion with using 0 for collapse-after.

So you could stay

```javascript
const shouldCollapse = collapseAt != 0 && category.pullRequests.length > collapseAt

if (shouldCollapse) {
  changeLog.push(
    categoryTitle,
    '\n\n',
    '<details>',
    '\n',
    `<summary>${category.pullRequests.length} changes</summary>`,
  )
} else {
  changeLog.push(categoryTitle, '\n\n', pullRequestString)
}
```

Reply:
Sure I can swap the `if`, but the problem with `0` as default value is that you cannot collapse all items. What if I want a category to always be collapsed?

Reply:
If you want it to always collapse, you use `collapse-after: 1` it would not collapse for one but for two PRs. I think that is a sensible behavior. Why collapse at exactly one item, you might as well just show that one item.

Reply:
In your scenario the default value should be `-1`, then if you define `collapse-after: 0` all items will be collapsed

Reply:
Also you would have to fix the plural of `number changes` in the summary.

Reply:
Valid point, let me update the PR :-)

Reply:
Note the rendering of the following markdown.

One shows that it does not make sense to always collapse if you only have 1 PR as it would take up the same size as two PRs
Two double news lines after summary matters for the rendering of the list 😄

### Deps

- Test

### Deps

<details>
<summary>1 changes</summary>

- Test

</details>

### Deps

<details>
<summary>1 changes</summary>
- Test
</details>

### Deps

<details>
<summary>2 changes</summary>

- Test
- Test

</details>

```
### Deps

- Test

### Deps

<details>
<summary>1 changes</summary>

- Test

</details>

### Deps

<details>
<summary>2 changes</summary>

- Test
- Test

</details>
```

Reply:
Yep, you're right :-) Checkout the latest code, i've also taken out the category title and pushed it to the changelog before determining the pull request string.